src/proxy.rs

```diff
@@ -237,6 +252,46 @@ impl RootContext for RootAuthThreescale {
     }
 }
 
+impl HttpAuthThreescale {
+    fn threescale_info_to_metadata(&self, ar: &authrep::AuthRep) -> Result<(), anyhow::Error> {
+        let apps = ar.apps();
+        let service = ar.service();
+        let usages = ar.usages();
+
+        if apps.is_empty() {
+            anyhow::bail!("could not extract application credentials");
+        }
+
+        if apps.len() > 1 {
+            debug!(
+                "found more than one source match for application - going to send {:?}",
+                apps[0]
+            );
+        }
+
+        let mut app_id_key = String::new();
+        let key_header = match &apps[0] {
+            Application::AppId(app_id, app_key) => {
+                app_id_key.push_str(app_id.as_ref());
+                app_id_key.push(':');
+                if let Some(key) = app_key {
+                    app_id_key.push_str(key.as_ref());
+                }
```

Comment:
I'd move the `:` to only be added to the string if there is actually a key following it.

Reply:
Right! We'll just have 1 key when splitting in the next filter if `app_key` is provided.